timemachine/cpp/CMakeLists.txt

```diff
@@ -9,7 +9,13 @@ project(timemachine LANGUAGES CXX CUDA)
 find_package(PythonInterp 3.7 REQUIRED)
 find_package(PythonLibs 3.7 REQUIRED)
 
-string(APPEND CMAKE_CUDA_FLAGS "-O3 -lineinfo")
+string(APPEND CMAKE_CUDA_FLAGS "-O3 -lineinfo -Werror all-warnings")
+string(APPEND CMAKE_CUDA_FLAGS
+  " -Wall"
+  " -Wno-sign-compare"
```

Comment:
I opted to disable `-Wsign-compare` for now, since it seems tricky to fix properly. It seems like we'd either need to transition to declaring `N`, `P`, `D`, etc. as `unsigned int`, or cast to unsigned wherever they are compared with `size_t`. The former would require changing the `Potential` interface.

Reply:
Yeah this is a nuisance - `static_casting` everywhere is sort of nuts (and just as unsafe).